### PR TITLE
[#3521] fix(client-python): Fix `hash()` function with multiple variables issue in `name_identitifer`

### PR DESCRIPTION
**Pull request by xloya**
### What changes were proposed in this pull request?

Fix `hash()` function in `name_identifier`.

### Why are the changes needed?

Fix: #3521 

### How was this patch tested?

Add a ut.

### CODE REVIEW

**Comment by xloya** [comment]
@xunliu Could you take a look of this? Thanks.

File: clients/client-python/tests/unittests/test_name_identifier.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    def test_name_identifier_hash(self):
+        name_identifier1: NameIdentifier = NameIdentifier.of_fileset(
+            "test_metalake", "test_catalog", "test_schema", "test_fileset1"
+        )
+        name_identifier2: NameIdentifier = NameIdentifier.of_fileset(
+            "test_metalake", "test_catalog", "test_schema", "test_fileset2"
+        )
+        identifier_dict = {name_identifier1: "test1", name_identifier2: "test2"}
+
+        self.assertEqual("test1", identifier_dict.get(name_identifier1))
```
[the remark this answers is not in the document]
Comment (xloya) [reply clients/client-python/tests/unittests/test_name_identifier.py:21]: Done

**Review by xunliu — APPROVED** [review]
LGTM

@xloya Thank you fix this issue.

**Review by jerryshao — APPROVED** [review]
LGTM. Thanks @xloya for your fix.

**Pull request merged**